lib/features/thread/data/network/thread_api.dart

```diff
@@ -17,14 +19,33 @@ import 'package:jmap_dart_client/jmap/mail/email/get/get_email_response.dart';
 import 'package:jmap_dart_client/jmap/mail/email/query/query_email_method.dart';
 import 'package:tmail_ui_user/features/thread/data/model/email_change_response.dart';
 import 'package:tmail_ui_user/features/thread/domain/model/email_response.dart';
+import 'package:tmail_ui_user/main/error/capability_validator.dart';
 
 class ThreadAPI {
 
   final HttpClient httpClient;
 
   ThreadAPI(this.httpClient);
 
+  Set<CapabilityIdentifier> _capabilitiesForEmailMethod(Session session, AccountId accountId) {
+    final getMailboxCreated = GetEmailMethod(accountId);
+    try {
+      requireCapability(
+          session,
+          accountId,
+          [CapabilityIdentifier.jmapTeamMailboxes]);
+      return {
+        CapabilityIdentifier.jmapCore,
+        CapabilityIdentifier.jmapMail,
+        CapabilityIdentifier.jmapTeamMailboxes
+      };
+    } catch (_) {
+      return getMailboxCreated.requiredCapabilities;
+    }
+  }
```

Comment:
move to `Capability_validator`. Please dont' use `final getMailboxCreated = GetEmailMethod(accountId);` in this method